internal/config/config.go

```diff
@@ -4,175 +4,100 @@ package config
-type restConfig struct {
-	Enabled bool   `yaml:"enabled" mapstructure:"enabled"`
-	Port    int    `yaml:"port" mapstructure:"port"`
-	Prefix  string `yaml:"prefix" mapstructure:"prefix"`
+// Database represents the juno database configuration.
+type Database struct {
+	Name string `yaml:"name" mapstructure:"name"`
+	Path string `yaml:"path" mapstructure:"path"`
 }
 
-// starknetConfig represents the juno StarkNet configuration.
-type starknetConfig struct {
-	Enabled       bool   `yaml:"enabled" mapstructure:"enabled"`
-	FeederGateway string `yaml:"feeder_gateway" mapstructure:"feeder_gateway"`
-	Network       string `yaml:"network" mapstructure:"network"`
-	ApiSync       bool   `yaml:"api_sync" mapstructure:"api_sync"`
+// Starknet represents the juno StarkNet configuration.
+type Starknet struct {
+	Enable    bool   `yaml:"enable" mapstructure:"enable"`
+	Sequencer string `yaml:"sequencer" mapstructure:"sequencer"`
+	Network   string `yaml:"network" mapstructure:"network"`
+	ApiSync   bool   `yaml:"apisync" mapstructure:"apisync"`
+	EthNode   string `yaml:"ethnode" mapstructure:"ethnode"`
 }
 
-// Config represents the juno configuration.
-type Config struct {
-	Logger   loggerConfig   `yaml:"logger" mapstructure:"logger"`
-	Ethereum ethereumConfig `yaml:"ethereum" mapstructure:"ethereum"`
-	RPC      rpcConfig      `yaml:"rpc" mapstructure:"rpc"`
-	Metrics  metricsConfig  `yaml:"metrics" mapstructure:"metrics"`
-	REST     restConfig     `yaml:"rest" mapstructure:"rest"`
-	DbPath   string         `yaml:"db_path" mapstructure:"db_path"`
-	Starknet starknetConfig `yaml:"starknet" mapstructure:"starknet"`
+// Juno is the top-level juno configuration.
+type Juno struct {
+	Log      Log      `yaml:"log" mapstructure:"log"`
+	Rpc      Rpc      `yaml:"rpc" mapstructure:"rpc"`
+	Metrics  Metrics  `yaml:"metrics" mapstructure:"metrics"`
+	Rest     Rest     `yaml:"rest" mapstructure:"rest"`
+	Database Database `yaml:"database" mapstructure:"database"`
+	Starknet Starknet `yaml:"starknet" mapstructure:"starknet"`
 }
 
-var (
-	// Dir is the default root directory for user-specific
-	// configuration data.
-	//
-	// On Darwin this is $HOME/Library/Application Support/juno/, on other
-	// Unix systems $XDG_CONFIG_HOME/juno/, and on Windows,
-	// %APPDATA%/juno.
-	Dir string
-	// DataDir is the is the default root directory for user-specific
-	// application data.
-	//
-	// On Unix this is $XDG_DATA_HOME/juno/ and on Windows,
-	// %APPDATA%/juno/.
-	DataDir string
-)
-
-// Runtime is the runtime configuration of the application.
-var Runtime *Config
-
-func init() {
-	// Set user config directory.
+// UserDataDir finds the user's default data directory, returning the
+// empty string and error otherwise.
+// It is analagous to os.UserConfigDir.
+func UserDataDir() (string, error) {
 	// notest
-	d, err := os.UserConfigDir()
-	if err != nil {
-		Logger.Fatal("Could not get config directory.")
-	}
-	Dir = filepath.Join(d, "juno")
+	junoDir := "juno"
```

Comment:
This doesn't look to ever be mutated afterwards to let's initialise it as a `const` for clarity.